### ISSUE TITLE
Issues Javax EE to Jakarta EE using java-saml

### ISSUE THREAD

**Issue by jacqueskpoty** [onelogin/java-saml#419]
It'is impossible to use java-saml with java 17 at this point. Is there any work around? is the community working on releasing a new version with java 17?

**Comment by haavar** [comment]
What's the issue with running under java 17?

**Comment by mauromol** [comment]
Java 17 itself has nothing to do with the change from Java EE to Jakarta EE.

**Comment by mauromol** [comment]
There's PR #395  open.

**Comment by william00179** [comment]
This package can be easily transformed with https://github.com/eclipse/transformer and then used.

I am using this for this package and many others that don't support the jakarta migration.

**Comment by haavar** [comment]
How are you guys using the servlet API with the SAML library? Can you not just pass in the parameters, instead of wrapping the request? I've been using jarkarta for a while now, and that did not affect how I used saml core.

**Comment by paulrutter** [comment]
Java-saml uses javax classes, java-saml-core does not. See https://github.com/SAML-Toolkits/java-saml#the-httprequest

**Comment by AbhishekGupta1996** [comment]
I’m currently working with the `java-saml` library and noticed that the `Auth `class expects parameters using `javax.*` packages. However, I’ve migrated my project to Java 17 and Jakarta EE, which uses `jakarta.*` packages, and I’m running into compatibility issues.

Has there been any update or support added for Jakarta EE in the `java-saml` dependency? If not, does anyone have a workaround or suggestion for using the `Auth `class with Jakarta EE?

Any guidance would be greatly appreciated.

Thanks!

**Comment by paulrutter** [comment]
> I’m currently working with the `java-saml` library and noticed that the `Auth `class expects parameters using `javax.*` packages. However, I’ve migrated my project to Java 17 and Jakarta EE, which uses `jakarta.*` packages, and I’m running into compatibility issues.
> 
> Has there been any update or support added for Jakarta EE in the `java-saml` dependency? If not, does anyone have a workaround or suggestion for using the `Auth `class with Jakarta EE?
> 
> Any guidance would be greatly appreciated.
> 
> Thanks!

See https://github.com/SAML-Toolkits/java-saml/issues/349#issuecomment-2020338432

**Comment by AbhishekGupta1996** [comment]
> > I’m currently working with the `java-saml` library and noticed that the `Auth `class expects parameters using `javax.*` packages. However, I’ve migrated my project to Java 17 and Jakarta EE, which uses `jakarta.*` packages, and I’m running into compatibility issues.
> > Has there been any update or support added for Jakarta EE in the `java-saml` dependency? If not, does anyone have a workaround or suggestion for using the `Auth `class with Jakarta EE?
> > Any guidance would be greatly appreciated.
> > Thanks!
> 
> See [#349 (comment)](https://github.com/SAML-Toolkits/java-saml/issues/349#issuecomment-2020338432)


Thanks @paulrutter  , that worked perfectly for me!
Quick question: is it safe to use wrapper conversions to javax.* in a real-time/production project?

**Comment by paulrutter** [comment]
I don't think it makes any difference. With the wrapper approach you do have both javax and jakarta apis in your classpath, which you wouldn't have with transformations.

**Comment by AbhishekGupta1996** [comment]
> I don't think it makes any difference. With the wrapper approach you do have both javax and jakarta apis in your classpath, which you wouldn't have with transformations.

Okay @paulrutter , Thanks for the update!!
